notebooks/contrib-dev/McCabe_Thiele_Steven_Final.ipynb

```diff
@@ -0,0 +1,1665 @@
+        "y_intercept_strip = xB * (1 - m_strip)\n",
+        "\n",
+        "# y-intercept of the rectifying opearting line\n",
+        "y_intercept_rec = xD * (1 - m_rec)\n",
+        "\n",
+        "### END SOLUTIONS"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "0dD5nKV6jjaD"
+      },
+      "source": [
+        "## 4.3. Generate the points needed to plot the stepping line, plot the McCabe-Thiele diagram, and calculate the number of stages\n",
+        "\n",
+        "Think where you want the \"stair\" to stop on your plot when you enter the y_start and y_end for the function.  "
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 13,
+      "metadata": {
+        "id": "FAIXZi5YVJwf"
+      },
+      "outputs": [],
+      "source": [
+        "### BEGIN SOLUTIONS\n",
+        "\n",
+        "# McCabe-Thiele of a feed run\n",
+        "# stripping portion of stepping line generation\n",
+        "stripping_line = stair(\n",
+        "    slope=m_strip,\n",
+        "    y_intercept=y_intercept_strip,\n",
+        "    x_start=xB,\n",
+        "    y_start=xB,\n",
+        "    y_end=intercept[1],\n",
+        ")\n",
+        "xplot_stripping = stripping_line[0]\n",
+        "yplot_stripping = stripping_line[1]\n",
+        "\n",
+        "# rectifying portion of stepping line generation\n",
+        "rectifying_line = stair(\n",
+        "    slope=m_rec,\n",
+        "    y_intercept=y_intercept_rec,\n",
+        "    x_start=(yplot_stripping[-1] - y_intercept_rec) / m_rec,\n",
+        "    y_start=yplot_stripping[-1],\n",
+        "    y_end=xD,\n",
+        ")\n",
+        "xplot_rectifying = rectifying_line[0]\n",
+        "yplot_rectifying = rectifying_line[1]\n",
+        "\n",
+        "# complie the x and y points to respective lists\n",
+        "xplot = xplot_stripping + xplot_rectifying\n",
+        "yplot = yplot_stripping + yplot_rectifying\n",
+        "\n",
+        "### END SOLUTIONS"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "### BEGIN SOLUTIONS\n",
+        "\n",
+        "# McCabe-Thiele diagram\n",
+        "fig = plt.figure(figsize=(4, 4))\n",
+        "# plot the LVE line\n",
+        "plt.plot(liqvapx, liqvapy)\n",
+        "# plot the 45 degree line\n",
+        "plt.plot([0, 1], [0, 1], color=\"orange\", linestyle=\":\")\n",
+        "# plot the step line\n",
+        "plt.plot(xplot, yplot, color=\"red\")\n",
+        "# plot the SOP\n",
+        "plt.plot([xB, intercept[0]], [xB, intercept[1]], color=\"purple\")\n",
+        "# plot the ROP\n",
+        "plt.plot([xD, intercept[0]], [xD, intercept[1]], color=\"green\")\n",
+        "# plot the q-line\n",
+        "plt.plot([xZ, intercept[0]], [xZ, intercept[1]], color=\"grey\")\n",
+        "# Formating the plot\n",
+        "plt.xlabel(\"Liquid Mole Fraction \\n of Methanol\", fontsize=10)\n",
```

Comment:
Change to "Methanol Liquid Mole Fraction" and "Methanol Vapor Mole Fraction". If needed, abbreviate "Mole Frac.". Increase font size per guidelines.